### PR TITLE
[SPARK-18690][PYTHON][SQL] Backward compatibility of unbounded frames

### PR DESCRIPTION
**Pull request by zero323**
## What changes were proposed in this pull request?

Makes `Window.unboundedPreceding` and `Window.unboundedFollowing` backward compatible.

## How was this patch tested?

Pyspark SQL unittests.

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by rxin** [comment]
Can you put a more descriptive title?

**Comment by zero323** [comment]
Sorry, long day. I was pretty sure I did.

File: python/pyspark/sql/tests.py [hunk 1]
```diff
@@ -1980,6 +1980,41 @@ def assert_runs_only_one_job_stage_and_task(job_group_name, f):
...
+                F.count("*").over(window.Window.rowsBetween(-sys.maxsize, sys.maxsize))
+            ).columns[0]
+
+        def range_frame_match():
+            return "RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING" in df.select(
+                F.count("*").over(window.Window.rangeBetween(-sys.maxsize, sys.maxsize))
+            ).columns[0]
+
+        with patch("sys.maxsize", 2 ** 31 - 1):
+            importlib.reload(window)
```
Comment (zero323) [inline python/pyspark/sql/tests.py]: This is something I don't like but it looks better than alternatives:
- Converting thresholds to methods so current `sys.maxsize` is picked up.
- Mocking thresholds as well.

**Comment by rxin** [comment]
Thanks - merging in master/branch-2.1.

**Pull request closed without merging**